infrastructure/src/main/java/org/corfudb/infrastructure/LogReplicationServer.java

```diff
@@ -0,0 +1,114 @@
+ * responsible of providing Log Replication across sites.
+ *
+ * The Log Replication Server, handles log replication entries--which
+ * represent parts of a Snapshot (full) sync or a Log Entry (delta) sync
+ * and also handles negotiation messages, which allows the Source Replicator
+ * to get a view of the last synchronized point at the remote site.
+ */
+@Slf4j
+public class LogReplicationServer extends AbstractServer {
+
+    private static final String configFilePath = "/config/corfu/log_replication_config.properties";
+
+    private final ServerContext serverContext;
+
+    private final ExecutorService executor;
+
+    @Getter
+    private final LogReplicationSinkManager sinkManager;
+
+    @Getter
+    private final HandlerMethods handler = HandlerMethods.generateHandler(MethodHandles.lookup(), this);
+
+    public LogReplicationServer(@Nonnull ServerContext context) {
+        this.serverContext = context;
+        this.executor = Executors.newFixedThreadPool(1,
+                new ServerThreadFactory("LogReplicationServer-", new ServerThreadFactory.ExceptionHandler()));
+        LogReplicationConfig config = getDefaultLogReplicationConfig();
+
+        // TODO (hack): where can we obtain the local corfu endpoint? site manager? or can we always assume port is 9000?
+        String corfuPort = serverContext.getLocalEndpoint().equals("localhost:9020") ? ":9001" : ":9000";
+        String corfuEndpoint = serverContext.getNodeLocator().getHost() + corfuPort;
+        log.info("Initialize Sink Manager with CorfuRuntime to {}", corfuEndpoint);
+        this.sinkManager = new LogReplicationSinkManager(corfuEndpoint, config);
+    }
+
+    private LogReplicationConfig getDefaultLogReplicationConfig() {
+        Set<String> streamsToReplicate = new HashSet<>(Arrays.asList("Table001", "Table002", "Table003"));
+        return new LogReplicationConfig(streamsToReplicate, UUID.randomUUID(), UUID.randomUUID());
+    }
+
+    /* ************ Override Methods ************ */
+
+    @Override
+    public boolean isServerReadyToHandleMsg(CorfuMsg msg) {
+        return getState() == ServerState.READY;
+    }
+
+    @Override
+    protected void processRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+        executor.submit(() -> getHandler().handle(msg, ctx, r));
+    }
+
+    @Override
+    public void shutdown() {
+        super.shutdown();
+        executor.shutdown();
+    }
+
+    /* ************ Server Handlers ************ */
+
+    @ServerHandler(type = CorfuMsgType.LOG_REPLICATION_ENTRY)
+    private void handleLogReplicationEntry(CorfuPayloadMsg<LogReplicationEntry> msg, ChannelHandlerContext ctx, IServerRouter r) {
+        log.info("Log Replication Entry received by Server.");
+
+        LogReplicationEntry ack = sinkManager.receive(msg.getPayload());
+
+        if (ack != null) {
+            long ts = ack.getMetadata().getMessageMetadataType().equals(MessageType.LOG_ENTRY_REPLICATED) ?
```

Comment:
what is the logic behind setting the timestamp vs snapshottimestamp?  can we add a comment to explain it?